miner/worker.go

```diff
@@ -551,64 +528,26 @@ func (w *worker) taskLoop() {
 func (w *worker) resultLoop() {
 	for {
 		select {
-		case block := <-w.resultCh:
-			// Short circuit when receiving empty result.
-			if block == nil {
-				continue
-			}
-			// Short circuit when receiving duplicate result caused by resubmitting.
-			if w.chain.HasBlock(block.Hash(), block.NumberU64()) {
+		case result := <-w.resultCh:
```

Comment:
Let's try to keep the code change as small as possible within this file.  So it's easier to merge upstream.

Reply:
Moved deep-copy part back to this file, `pendingTasks` is removed.